### ISSUE TITLE
User is not present when using client-credentials grant type on DRF + django-oauth-toolkit

### ISSUE THREAD

**Issue by nicholasamorim** [jrobichaud/django-structlog#56]
A common pattern between service-to-service authentication is to use OAuth2 with what's called "client credentials".

Those are supported by DRF + [Django-oauth-toolkit](https://django-oauth-toolkit.readthedocs.io/en/latest/index.html).

The issue here is that there's no _user_ per se. DRF does authenticate the request but it's not tied to an actual `User`, so what happens is that `bind_user_id` fails with `'NoneType' object has no attribute 'pk'`

```python
@staticmethod
def bind_user_id(request):
    if hasattr(request, "user"):
        user_id = request.user.pk
        if isinstance(user_id, uuid.UUID):
            user_id = str(user_id)
        logger.bind(user_id=user_id)
```

I think this is slightly related to #37 at least. 

What's an accepted workaround here? The check merely checks for the _existence_ of the variable user but not whether there's an actual value in it.

**Comment by jrobichaud** [comment]
I'll fix for tomorrow. Thank you for your patience.

**Comment by jrobichaud** [comment]
Fixed in 2.0.0

**Comment by nicholasamorim** [comment]
Just commenting to say thank you very much!
